polars/polars-core/src/chunked_array/temporal/datetime.rs

```diff
@@ -161,7 +170,19 @@ impl DatetimeChunked {
                     None => mutarr.push_null(),
                     Some(v) => {
                         buf.clear();
-                        let datefmt = conversion_f(*v).format(fmt);
+                        let converted = conversion_f(*v);
```

Comment:
This is an optimization we could add, it is not a blocker for this PR, but if you feel up to it, you can try to implement it this way.

We can make put `conversion_f` behind a reference and pass a different `conversion_f` function dependend of the timezone on the array. That means that we don't branch on timezone every iteration, but only once when we set the `conversion_f` we will use for this function.

Reply:
good one, thanks for your review!

OK, giving this a go (I've made the other change in the meantime anyway)

Reply:
I've tried factoring the function out, is this what you meant?

I'm running a timing test now to check the performance impact, will post the results when it's done

Reply:
If I've done this right, then it doesn't look like there's any performance difference

Experiments notebook: https://www.kaggle.com/code/marcogorelli/polars-timing?scriptVersionId=118296517

Previous commit:
- 475 ms ± 1.58 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
- 491 ms ± 16.6 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
- 481 ms ± 12.3 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

Most recent commit (with the function factored out):
- 481 ms ± 1.54 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
-495 ms ± 16.9 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
-485 ms ± 6.43 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

Here's the script I ran (both times with `make build-release`):
```python
from IPython import get_ipython
from datetime import datetime
import polars as pl

ipython = get_ipython()


dr = pl.date_range(datetime(1000, 1, 1), datetime(2000, 1, 1), interval='6h', time_zone='US/Pacific')

print(ipython.run_line_magic("timeit", "dr.dt.strftime('%z')"))
print(ipython.run_line_magic("timeit", "dr.dt.strftime('%z')"))
print(ipython.run_line_magic("timeit", "dr.dt.strftime('%z')"))
```

Reply:
I ran this on a larger input and got consistently better results with the previous commit
https://www.kaggle.com/code/marcogorelli/polars-timing/notebook
I'll revert tomorrow then. If you were thinking of merging, please wait - thanks! 🙏